### ISSUE TITLE
Code review for E6 module work

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/chipper#872]
> (SR, CM, JB? code review, 4-8 hours) [sprintable, pre-publication] Code quality for modulify and the preprocessed modules, the revisions in the chipper build, and the webpack file: Much of the code is prototype-oriented. This is fine for tasks like grunt migrate which will ultimately be deleted after we run it once. But for grunt modulify and the JS harnesses for images, audio, text, etc should be brought to production standards.
Inline templates in modulify should be top level template files
JO: I would prefer `grunt modulify` would just change the repo that it is run in, and it should be included as part of `grunt update`

Once things are more solidified, we should review all changed chipper files and make sure they are up to our standards.

**Comment by pixelzoom** [comment]
Deferred until there's something specific to review.

**Comment by jbphet** [comment]
@jonathanolson 

> Once things are more solidified, we should review all changed chipper files and make sure they are up to our standards.

This issue is assigned to me, but I'm not clear on what I need to do to address it.  It almost sounds like it's not quite ready for review.  @jonathanolson - can you please clarify?

**Comment by jonathanolson** [comment]
It sounds like it's not ready yet, presumably this would be done a bit more during the "sprint"? It's unclear and very broad to me.

**Comment by chrisklus** [comment]
Code to review:

in `joist`:

- [ ] SimLauncher.js

in `chipper`:

- [ ] buildRunnable.js
- [ ] buildStandalone.js
- [ ] generateDevelopmentHTML.js
- [ ] generateTestHTML.js
- [ ] getInitializationScript.js
- [ ] getStringMap.js
- [ ] Gruntfile.js
- [ ] modulify.js
- [ ] sortImports.js
- [ ] wepackBuild.js
- [ ] getStringModule.js
- [ ] load-runtime-strings.js
- [ ] sim-development.html
- [ ] sim-webpack.ejs

**Comment by pixelzoom** [comment]
@ariel-phet Before someone(s) are assigned to code review, we should confirm that the authors have done a review themselves using the CRC.

**Comment by jonathanolson** [comment]
reportUnusedMedia/reportUnusedStrings are commented out in buildRunnable, otherwise I believe the code I wrote is ready to review.

**Comment by pixelzoom** [comment]
3/5/20 dev meeting

* List in https://github.com/phetsims/chipper/issues/872#issuecomment-591604280 is probably complete.

* @samreid needs to self review modify.js first.  Assigning issue to him to do this, then back to dev meeting to divvy up review.

**Comment by mattpen** [comment]
3/19 dev meeting:

@samreid still needs to self-review modulify.js

@Denz1994 will start review of buildRunnable.js, webpackBuild.js, and Gruntfile.js.

@jonathanolson said that most recent changes should be g2g, but there may be old changes that are not up to standards.

**Comment by samreid** [comment]
modulify.js is now ready for code review.  Please note I added a few questions for the reviewer in comments.

**Comment by Denz1994** [comment]
I reviewed the code that was relevant to `webpack` in `buildRunnable.js` `chipper/gruntfile.js` and `webpackBuild.js`. There were only a few suggestions in the  `REVIEW` comments added (nothing major). The `grunt webpack-dev-server` help doc looks good and all the options documented were tested. I will reassign to @jonathanolson to address the comments.

**Comment by jbphet** [comment]
Discussed at 3/26/2020 dev meeting, next steps are for @jonathanolson to address the comments added by @Denz1994, and @zepumph is going to review string-related files.  @Denz1994 is going to review `sortImports` this week as well.  We will revisit next week and assign more.

**Comment by jonathanolson** [comment]
@samreid I'm curious about your opinions on the remaining 2 REVIEW notes in chipper. Notably:

1. What should we do when mipmap creation fails? Erroring out makes sense, but this was commented out with a try-catch it appears?
2. Should modulifyFile (in modulify.js) be simplified with logic? This looks fine to leave as-is for me.

**Comment by samreid** [comment]
Mipmaps failed for some images, via the interlacing pattern described in https://github.com/phetsims/chipper/issues/831.  I wasn't sure how throw interplayed with `async` so I tried removing the try/catch and adding a manual test `throw new Error()` and it had the desired behavior.  I'll commit this change.

**Comment by samreid** [comment]
> Should modulifyFile (in modulify.js) be simplified with logic? This looks fine to leave as-is for me.

Thanks, I removed those comments.

**Comment by Denz1994** [comment]
Review comments for sortImports.js have been added to https://github.com/phetsims/chipper/issues/930.

**Comment by Denz1994** [comment]
From Dev Meeting 04/09/2020:

I'll review `generateDevelopmentHTML.js` and `generateTestHTML.js` this week. @jonathanolson Will respond to review statements in sortImports.js.

**Comment by Denz1994** [comment]
I've started reviewing getInitializationScript.js and buildStandAlone.js. I'll add SimLauncher.js as a priority as well. These files should be reviewed by Friday.

**Comment by samreid** [comment]
SimLauncher has many changes in a branch, see https://github.com/phetsims/joist/issues/621 that will be moved to master soon.

**Comment by zepumph** [comment]
From dev meeting today, @Denz1994 will wait on reviewing SimLauncher.js until @samreid merges those changes.

**Comment by samreid** [comment]
I addressed review feedback and moved SimLauncher to master.

**Comment by Denz1994** [comment]
The last file to review is modulify.js. I'll get to it this week to finish off this chip away.

**Comment by Denz1994** [comment]
All the files listed in https://github.com/phetsims/chipper/issues/872#issuecomment-591604280 were reviewed or have their own issues.

**Comment by jonathanolson** [comment]
Looks like everything is in remaining other issues, closing here. Thanks!
